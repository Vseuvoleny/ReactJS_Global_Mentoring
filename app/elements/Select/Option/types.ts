export interface IOption {
    options: any[],
    name: string
}