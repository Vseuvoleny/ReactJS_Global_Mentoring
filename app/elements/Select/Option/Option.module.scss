@import "../../../styles/colors/colors.scss";

.option {
  position: relative;
  &-checkbox {
    opacity: 0;
    position: absolute;
    z-index: -1;
  }
  &_label {
    width: 100%;
    display: inline-block;
    padding: 6px 12px;
    cursor: pointer;
  }

  &_checkbox:checked + &_label {
    color: $color-corral;
  }
}
