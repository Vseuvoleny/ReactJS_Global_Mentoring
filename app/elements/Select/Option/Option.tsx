import React, { FC } from "react";
import { Field } from "formik";
import classes from "./Option.module.scss";
import { IOption } from "./types";

const Option: FC<IOption> = ({ options, name }) => {
  return (
    <div className={classes.option}>
      {options.map((option) => {
        return (
          <label key={option.id} className={classes.option_label}>
            <Field
              type="checkbox"
              name={name}
              value={option.title}
              className={classes.option_checkbox}
            />
            {option.title}
          </label>
        );
      })}
    </div>
  );
};

export default Option;
