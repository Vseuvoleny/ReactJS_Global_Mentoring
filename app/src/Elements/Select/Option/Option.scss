@import "../../../styles/colors/colors.scss";

.option {
  position: relative;
  &-checkbox {
    opacity: 0;
    position: absolute;
    z-index: -1;
  }
  &-label {
    width: 100%;
    display: inline-block;
    padding: 6px 12px;
    cursor: pointer;
  }

  &-checkbox:checked + &-label {
    color: $color-corral;
  }
}
