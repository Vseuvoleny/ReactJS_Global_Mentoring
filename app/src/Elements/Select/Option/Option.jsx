import React from "react";
import PropTypes from "prop-types";
import { Field } from "formik";
import "./Option.scss";

const Option = ({ options, name }) => {
  return (
    <div className="option">
      {options.map((option) => {
        return (
          <label key={option.id} className="option-label">
            <Field
              type="checkbox"
              name={name}
              value={option.title}
              className="option-checkbox"
            />
            {option.title}
          </label>
        );
      })}
    </div>
  );
};

Option.propTypes = {
  options: PropTypes.array,
  name: PropTypes.string,
};

export default Option;
