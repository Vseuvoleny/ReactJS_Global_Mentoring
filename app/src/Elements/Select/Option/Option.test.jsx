import { render } from "@testing-library/react";
import { Formik } from "formik";
import Option from "./Option";
import React from "react";

describe("Option component", () => {
  test("should render Option component with a list", () => {
    const fakeOptions = [
      { title: "title_1", id: 1 },
      { title: "title_2", id: 2 },
      { title: "title_3", id: 3 },
      { title: "title_4", id: 4 },
    ];

    const { container } = render(
      <Formik>
        <Option options={fakeOptions} />
      </Formik>
    );

    expect(container).toMatchSnapshot();
  });
});
