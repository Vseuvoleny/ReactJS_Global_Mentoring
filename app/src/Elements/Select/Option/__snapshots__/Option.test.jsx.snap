// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option component should render Option component with a list 1`] = `
<div>
  <div
    class="option"
  >
    <label
      class="option-label"
    >
      <input
        class="option-checkbox"
        type="checkbox"
        value="title_1"
      />
      title_1
    </label>
    <label
      class="option-label"
    >
      <input
        class="option-checkbox"
        type="checkbox"
        value="title_2"
      />
      title_2
    </label>
    <label
      class="option-label"
    >
      <input
        class="option-checkbox"
        type="checkbox"
        value="title_3"
      />
      title_3
    </label>
    <label
      class="option-label"
    >
      <input
        class="option-checkbox"
        type="checkbox"
        value="title_4"
      />
      title_4
    </label>
  </div>
</div>
`;
